packages/legend-application-query/src/components/QueryEditor.tsx

```diff
@@ -819,6 +819,114 @@ const QueryEditorHeaderContent = observer(
   },
 );
 
+const QueryEditorExistingQueryVersionRevertModal = observer(
+  (props: { existingEditorStore: ExistingQueryEditorStore }) => {
+    const { existingEditorStore } = props;
+    const updateState = existingEditorStore.updateState;
+    const applicationStore = existingEditorStore.applicationStore;
+    const [queryVersion, setQueryVersion] = useState<string | undefined>(
+      existingEditorStore.lightQuery.originalVersionId,
+    );
+    const versionOptions = [
+      LATEST_VERSION_ALIAS,
+      ...updateState.projectVersions,
+    ]
+      .slice()
+      .sort((v1, v2) => compareSemVerVersions(v2, v1))
+      .map(buildVersionOption);
+    const selectedVersionOption = queryVersion
+      ? buildVersionOption(queryVersion)
+      : null;
+    const onVersionOptionChange = async (
+      option: VersionOption | null,
+    ): Promise<void> => {
+      if (option?.value && option.value !== queryVersion) {
+        setQueryVersion(option.value);
+      }
+    };
+    const updateQueryVersionId = applicationStore.guardUnhandledError(
+      async (): Promise<void> => {
+        if (queryVersion) {
+          flowResult(
+            existingEditorStore.updateState.updateQueryVersionId(
```

Comment:
How come here we use `existingEditorStore.updateState` but on line 857 we just use `updateState`?

Reply:
loll you are right thanks resolved